extensions/circuit/src/adapters/allocate_frame.rs

```diff
@@ -174,6 +193,7 @@ impl<F: PrimeField32> VmAdapterChipWom<F> for AllocateFrameAdapterChipWom {
 
         self.next_fp += amount_bytes;
 
+        // TODO: "next fp" is being handled as a register/mem read (i.e., a read in the `BasicAdapterInterface`), is this right?
```

Comment:
no don't think so, it should be a nondeterministic value from the prover, so we just keep it in the struct here

Reply:
you mean in `AllocateFrameReadRecord`?